scripts/o3de/tests/test_repo_properties.py

```diff
@@ -540,3 +541,54 @@ def mock_dryrun_print(msg, *args, **kwargs):
                             assert template in str(JSON_DICT.get('template_json_key'))
                         else:
                             assert template in expected_templates
+
+    # test copy_to_repo_subfolder
+    @pytest.mark.parametrize("repo_path, repo_name, add_gems, copy_to_repo_subfolder, \
+                            expected_result", [
+        # test copy a gem to repo subfolder
+        pytest.param(pathlib.Path('F:/repotest'), 'copytest1', [JSON_DICT['gem_archive_json_key']], temp_folder, 0)
+        
+    ])
+    def test_copy_to_repo_subfolder(self, temp_folder, repo_path, repo_name, add_gems, copy_to_repo_subfolder,
+                                    expected_result):
+        def get_repo_props(repo_path: str or pathlib.Path = None) -> dict or None:
+            if not repo_path:
+                self.repo_json.data = None
+                return None
+            return self.repo_json.data
+        
+        def save_o3de_manifest(json_data: dict, manifest_path: pathlib.Path = None) -> bool:
+            self.repo_json.data = json_data
+            return True
+        
+        def backup_file(file_name: str or pathlib.Path) -> None:
+            self.backup_file_name = file_name
+        
+        object_paths = [
+            temp_folder / 'gem'
+        ]
```

Comment:
I'm not sure why you need this array?  could you just remove it and use `temp_folder / 'gem'` on line 585?
```suggestion
        object_paths = [
            temp_folder / 'gem'
        ]
```

Reply:
The original functions takes a list object and splits it into path, so I have to pass it in as a list

Reply:
It that case I think you should remove `add_gems` if that input is never used and just use this `object_paths` variable.